app/views/tasks/show.html.erb

```diff
@@ -80,7 +83,43 @@
     Tag: <%= @task.tag %>
   </div>
 
-  <% if current_user.role == 'admin' || current_user.role == 'moderator' %>
+  <% if @task.claimed_by == nil %>
+    <div>
+      <%= form_with model: @task do |form| %>
+
+      <div>
+        <%= form.hidden_field(:claimed_by, value: current_user.id) %>
+        <% @task.errors.full_messages_for(:claimed_by).each do |message| %>
+        <div><%= message %></div>
+        <% end %>
+      </div>
+
+      <div>
+        <%= form.button :"Claim Task" %>
+      </div>
+      <% end %>
+    </div>
+  <% end %>
+
+  <% if @task.claimed_by == current_user.id %>
+    <div>
+      <%= form_with model: @task do |form| %>
+
+      <div>
+        <%= form.hidden_field(:claimed_by, value: nil) %>
+        <% @task.errors.full_messages_for(:claimed_by).each do |message| %>
+        <div><%= message %></div>
+        <% end %>
+      </div>
+
+      <div>
+        <%= form.button :"Unclaim Task" %>
+      </div>
+      <% end %>
+    </div>
+  <% end %>
+
+  <% if current_user.role == 'admin' || current_user.role == 'Gamer' %>
```

Comment:
Gamer? lol. You are creating a backdoor or what? Anyone could make a role titled "Gamer"